rflx/model.py

```diff
@@ -300,40 +426,57 @@ def valid_message_field_types(message: "AbstractMessage") -> bool:
     return True
 
 
+class MessageState(Base):
+    fields: Optional[Tuple[Field, ...]] = ()
+    paths: Mapping[Field, Set[Tuple[Link, ...]]] = {}
+    definite_predecessors: Mapping[Field, Tuple[Field, ...]] = {}
+    field_condition: Mapping[Field, Expr] = {}
+
+
 @invariant(lambda self: valid_message_field_types(self))
 class AbstractMessage(Type):
+    # pylint: disable=too-many-arguments
     def __init__(
-        self, identifier: StrID, structure: Sequence[Link], types: Mapping[Field, Type]
+        self,
+        identifier: StrID,
+        structure: Sequence[Link],
+        types: Mapping[Field, Type],
+        location: Location = None,
+        error: RecordFluxError = None,
+        state: MessageState = None,
     ) -> None:
-        super().__init__(identifier)
+        super().__init__(identifier, location, error)
 
         self.structure = structure
         self.__types = types
-
-        if structure or types:
-            self.__verify()
-            self.__fields = self.__compute_topological_sorting()
-            self.__types = {f: self.__types[f] for f in self.__fields}
-            self.__paths = {f: self.__compute_paths(f) for f in self.all_fields}
-            self.__definite_predecessors = {
-                f: self.__compute_definite_predecessors(f) for f in self.all_fields
-            }
-            self.__field_condition = {
-                f: self.__compute_field_condition(f).simplified() for f in self.all_fields
-            }
-            self.__verify_conditions()
-        else:
-            self.__fields = ()
-            self.__paths = {}
-            self.__definite_predecessors = {}
-            self.__field_condition = {}
+        self.__has_unreachable = False
+        self._state = state or MessageState()
+
+        if not state and (structure or types):
+            try:
+                self.__verify()
+                self._state.fields = self.__compute_topological_sorting()
+                if self._state.fields:
+                    self.__types = {f: self.__types[f] for f in self._state.fields}
+                    self._state.paths = {f: self.__compute_paths(f) for f in self.all_fields}
+                    self._state.definite_predecessors = {
+                        f: self.__compute_definite_predecessors(f) for f in self.all_fields
+                    }
+                    self._state.field_condition = {
+                        f: self.__compute_field_condition(f).simplified() for f in self.all_fields
+                    }
+                    self.__verify_conditions()
+            except RecordFluxError:
+                pass
```

Comment:
`RecordFluxError` may be raised e.g. in `__verify_condition`. However, we don't want to raise it in the message constructor, as this would end parsing after the first error and we couldn't collect other errors (in other constructors during parsing or later during model validation).

Reply:
Isn't the information in this caught `RecordFluxError` lost in this case?

Reply:
No, the error information is appended to `self.error` of the message instance.